source/uwp/Renderer/lib/AdaptiveActionInvoker.cpp

```diff
@@ -3,39 +3,16 @@
 #include "pch.h"
 
 #include "AdaptiveActionInvoker.h"
+#include "AdaptiveActionInvoker.g.cpp"
 #include "Util.h"
 
-using namespace Microsoft::WRL;
-using namespace Microsoft::WRL::Wrappers;
-using namespace ABI::AdaptiveCards::Rendering::Uwp;
-using namespace ABI::AdaptiveCards::ObjectModel::Uwp;
-using namespace ABI::Windows::Foundation;
-using namespace ABI::Windows::UI::Xaml;
-
-namespace AdaptiveCards::Rendering::Uwp
+namespace winrt::AdaptiveCards::Rendering::Uwp::implementation
 {
-    HRESULT AdaptiveActionInvoker::RuntimeClassInitialize() noexcept { return S_OK; }
-
-    HRESULT AdaptiveActionInvoker::RuntimeClassInitialize(_In_ RenderedAdaptiveCard* renderResult) noexcept
-    try
-    {
-        ComPtr<RenderedAdaptiveCard> strongRenderResult = renderResult;
-        return strongRenderResult.AsWeak(&m_weakRenderResult);
-    }
-    CATCH_RETURN;
-
-    HRESULT AdaptiveActionInvoker::SendActionEvent(_In_ IAdaptiveActionElement* actionElement)
+    void AdaptiveActionInvoker::SendActionEvent(rtom::IAdaptiveActionElement const& actionElement)
     {
-        ComPtr<IRenderedAdaptiveCard> strongRenderResult;
-        RETURN_IF_FAILED(m_weakRenderResult.As(&strongRenderResult));
-        if (strongRenderResult != nullptr)
+        if (auto strong = peek_innards<implementation::RenderedAdaptiveCard>(m_weakRenderResult.get()))
```

Comment:
what's peek_innards? Are they the replacement for 
get_weak/get_strong in cppwinrt?

Reply:
@licanhua it's like **winrt::get_self** but with checking the struct **guid** so it matches our implementation.